### PR TITLE
Add time tracker to ensure at most one sync occurs in a minute

### PR DESCRIPTION
**Pull request by bettysunx13**
Now user will be limited to initiate at most one synchronization every minute.

### CODE REVIEW

**Comment by danielqiang** [comment]
LGTM!

File: src/background.ts [hunk 1]
```diff
@@ -6,15 +6,34 @@
...
+
+    chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
+        if (request.action === "oauth") {
+            autocalendar.calendar.session.oauth_token();
+        } else if (request.action === "sync_canvas") {
+            let time_diff =
+                prev_execution === null
+                    ? Number.MAX_VALUE
+                    : new Date().getTime() - prev_execution.getTime();
+            if (!isRunning && time_diff > 62000) {
```
Comment (danielqiang) [inline src/background.ts]: This should be 60 seconds, leaving a 2 second buffer seems pointless to me

File: src/background.ts [hunk 2]
```diff
@@ -6,15 +6,34 @@
...
+                isRunning = true;
                 console.time("sync_canvas");
-                await autocalendar.sync_canvas(calendar_name);
-                console.timeEnd("sync_canvas");
-            } else if (request.action == "oauth") {
-                await autocalendar.calendar.session.oauth_token();
+                autocalendar.sync_canvas(calendar_name).then(() => {
+                    console.timeEnd("sync_canvas");
```
Comment (danielqiang) [inline src/background.ts]: console.timeEnd is to time the function, it doesn't make sense to interleave it with the rest of the code. We should probably move it to the end of the function after sendResponse().

File: src/background.ts [hunk 3]
```diff
@@ -6,15 +6,34 @@
     const calendar_name = "AutoCalendar Demo";
     const autocalendar = new AutoCalendar();
 
-    chrome.runtime.onMessage.addListener(
-        async (request, sender, sendResponse) => {
-            if (request.action === "sync_canvas") {
+    let prev_execution = null;
+    let isRunning = false;
```
Comment (danielqiang) [inline src/background.ts:10]: This flag is almost certainly unnecessary; you should be able to refactor it out. Flags are only necessary within complex algorithms usually, they clutter up the code a lot so should be avoided when possible.

File: src/background.ts (hunk 1 above)
Comment (bettysunx13) [reply src/background.ts]: Solved!

File: src/background.ts (hunk 2 above)
Comment (bettysunx13) [reply src/background.ts]: Sounds good.

File: src/background.ts (hunk 3 above)
Comment (bettysunx13) [reply src/background.ts:10]: Currently there is no better way to ensure one sync is happening at a time, and the flag is required by the front-end to behave properly depending on if background is running or not.

**Pull request merged**